### PR TITLE
Function file_get_contents has been replaced with humbug_get_contents

### PR DESCRIPTION
**Pull request by msojda**
Fixes puli/issues#138

### CODE REVIEW

File: src/Installer.php [hunk 1]
```diff
@@ -56,7 +56,6 @@ class Installer
 --version="..."      install a specific version
 --filename="..."     set the target filename (default: puli.phar)
 --disable-tls        disable SSL/TLS security for file downloads
---cafile="..."       set the path to a Certificate Authority (CA) certificate file for SSL/TLS verification
```
Comment (tgalopin) [inline src/Installer.php:59]: This breaks BC, I think we should add it in the Changelog.

Comment (webmozart) [inline src/Installer.php:59]: Is there no way to set the cafile with `humbug_get_contents()`?

Comment (tgalopin) [inline src/Installer.php:59]: It doesn't seems so, at least not without extending it: https://github.com/padraic/file_get_contents/blob/master/src/Humbug/FileGetContents.php#L28

But I'm not sure.

Comment (msojda) [inline src/Installer.php:59]: @tgalopin is right, unfortunately there is no way without extending

Comment (webmozart) [inline src/Installer.php:59]: Alright, let's keep it this way then.

File: src/HttpClient.php [hunk 2]
```diff
@@ -437,14 +159,11 @@ private function getMergedStreamContext($url)
...
 
         if (isset($options['http']['header'])) {
-            $options['http']['header'] .= "Connection: close\r\n";
+            $options['http']['header'] .= implode('', $this->headers);
         } else {
-            $options['http']['header'] = "Connection: close\r\n";
+            $options['http']['header'] = implode('', $this->headers);
         }
-        if (extension_loaded('zlib')) {
-            $options['http']['header'] .= "Accept-Encoding: gzip\r\n";
```
Comment (webmozart) [inline src/HttpClient.php]: Why has this been removed?

File: src/HttpClient.php [hunk 3]
```diff
@@ -289,14 +60,23 @@ public function __construct($disableTls = false, $cafile = null)
      */
     public function download($url)
     {
-        $context = $this->getStreamContext($url);
-        $result = file_get_contents($url, null, $context);
+        if (extension_loaded('zlib')) {
+            array_push($this->headers, "Accept-Encoding: gzip\r\n");
```
Comment (webmozart) [inline src/HttpClient.php]: You can simply do `$this->headers[] = ...` here, I think that's clearer.

Comment (webmozart) [inline src/HttpClient.php]: Shouldn't this be done in the constructor? Right now, for every call to `download()` the header will be added again.

File: src/HttpClient.php [hunk 4]
```diff
@@ -289,14 +60,23 @@ public function __construct($disableTls = false, $cafile = null)
...
+        }
+
+        if ($this->tls) {
+            humbug_set_headers($this->headers);
+            $result = humbug_get_contents($url);
+        } else {
+            $context = $this->getNonSslStreamContext($url);
+            $result = file_get_contents($url, null, $context);
```
Comment (webmozart) [inline src/HttpClient.php]: I think we are able to remove this block too. All this code is handled inside `humbug_get_contents()` too.

Comment (msojda) [inline src/HttpClient.php]: We were talking about that with @tgalopin yesterday. If I remove that block we'll be unable to handle `--disable-tls` option. In that case `file_get_contents` has to have it's own context with proxy stuff configured on our side. Does it make sense?

**Comment by tgalopin** [comment]
This seems fine for me, can you squash the commits please? Thanks for your work!

@webmozart what do you think of this?

File: src/HttpClient.php (hunk 4 above)
Comment (webmozart) [inline src/HttpClient.php]: I think we can remove the `--disable-tls` option for now. Let's do this once it becomes a problem.

Comment (msojda) [inline src/HttpClient.php]: Removed :)

**Comment by webmozart** [comment]
Looks good, thanks a lot @msojda! :)

**Pull request merged**